pkg/util/test/util.go

```diff
@@ -330,6 +365,57 @@ func ValidateStorageCluster(
 	return nil
 }
 
+func validateStorageNodes(pxImageList map[string]string, cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
+	var pxVersion string
+
+	// Construct PX Version string used to match to deployed expected PX version
+	if strings.Contains(pxImageList["version"], "_") {
+		if len(cluster.Spec.CommonConfig.Env) > 0 {
+			for _, env := range cluster.Spec.CommonConfig.Env {
+				if env.Name == PxReleaseManifestURLEnvVarName {
+					pxVersion = strings.TrimSpace(regexp.MustCompile(`\S+\/(\S+)\/version`).FindStringSubmatch(env.Value)[1])
+					if pxVersion == "" {
+						return fmt.Errorf("failed to extract version from value of %s", PxReleaseManifestURLEnvVarName)
+					}
+				}
+			}
+		}
+	} else {
+		pxVersion = strings.TrimSpace(regexp.MustCompile(`:(\S+)`).FindStringSubmatch(pxImageList["version"])[1])
+	}
+
+	t := func() (interface{}, bool, error) {
+		// Get all StorageNodes
+		storageNodeList, err := operatorops.Instance().ListStorageNodes(cluster.Namespace)
+		if err != nil {
+			return nil, true, err
+		}
+
+		// Check StorageNodes status and PX version
+		expectedStatus := "Online"
+		var readyNodes int
+		for _, storageNode := range storageNodeList.Items {
+			if storageNode.Status.Phase == expectedStatus && strings.Contains(storageNode.Spec.Version, pxVersion) {
+				readyNodes++
+				logrus.Debugf("storagenode: %s, Expected status: %s Got: %s, Expected PX version: %s Got: %s", storageNode.Name, expectedStatus, storageNode.Status.Phase, pxVersion, storageNode.Spec.Version)
+			} else {
+				logrus.Debugf("storagenode: %s Expected status: %s Got: %s, Expected PX version: %s Got: %s", storageNode.Name, expectedStatus, storageNode.Status.Phase, pxVersion, storageNode.Spec.Version)
```

Comment:
Both the debug statements are identical. We can just remove the else and put it after the if condition.

Reply:
Done